src/health.rs

```diff
@@ -0,0 +1,67 @@
+use std::time::Duration;
+
+use futures::*;
+use tokio_timer::*;
+use tokio_core::reactor::Core;
+use hyper::{Client, Url};
+
+use pool::Pool;
+
+pub struct HealthCheck {
+    interval: Duration,
+    pool: Pool,
+    uri_path: String,
+}
+
+impl HealthCheck {
+    pub fn new(interval: Duration, pool: Pool, uri_path: String) -> HealthCheck {
+
+        HealthCheck {
+            interval: interval,
+            pool: pool,
+            uri_path: uri_path,
+        }
+    }
+
+    pub fn run(&self) {
+        let mut core = Core::new().unwrap();
+        let timer = Timer::default();
+
+        let handle = core.handle();
+        let client = Client::new(&handle);
+
+        let pool = self.pool.clone();
+        let work = timer.interval(self.interval).for_each(move |()| {
+            let servers = pool.all();
+            for server in servers {
+                let url = format!("http://{}{}", server, self.uri_path);
+                let url = Url::parse(&url).unwrap();
+                debug!("Health check {:?}", url);
+
+                let pool1 = pool.clone();
+                let pool2 = pool.clone();
+                let req = client.get(url).and_then(move |res| {
+                    debug!("Response: {}", res.status());
+                    debug!("Headers: \n{}", res.headers());
+
+                    if ! res.status().is_success() {
+                        info!("Removing {} from pool", server);
```

Comment:
Maybe you could add an explanation `. Reason: a GET request on {} return the status code {} and response body {}` ?